manta-trusted-setup/src/groth16/ceremony/config/ppot.rs

```diff
@@ -308,30 +325,65 @@ pub fn register(twitter_account: String, email: String) {
 
 /// Prompts the client information and get client keys.
 #[inline]
-pub fn get_client_keys() -> Option<(ed25519::SecretKey, ed25519::PublicKey)> {
+pub fn get_client_keys() -> Result<(ed25519::SecretKey, ed25519::PublicKey), ClientKeyError> {
     println!(
         "Please enter the {} you received when you registered yourself using this tool.",
         "Secret".italic()
     );
-    let seed_bytes = Seed::new(
-        &Mnemonic::from_phrase(
-            Input::with_theme(&ColorfulTheme::default())
-                .with_prompt("Your Secret")
-                .validate_with(|input: &String| -> Result<(), &str> {
-                    Mnemonic::validate(input, Language::English)
-                        .map_err(|_| "This is not a valid secret.")
-                })
-                .interact_text()
-                .expect("Please enter your secret received during `Register`.")
-                .as_str(),
-            Language::English,
-        )
-        .expect("Should produce a mnemonic from the secret."),
-        "manta-trusted-setup",
-    )
-    .as_bytes()
-    .to_vec();
-    generate_keys(&seed_bytes)
+    let text = match Input::with_theme(&ColorfulTheme::default())
+        .with_prompt("Your Secret")
+        .validate_with(|input: &String| -> Result<(), &str> {
+            Mnemonic::validate(input, Language::English).map_err(|_| "This is not a valid secret.")
+        })
+        .interact_text()
+    {
+        Ok(text) => text,
+        Err(_) => return Err(ClientKeyError::InvalidSecret),
+    };
+    let mnemonic = match Mnemonic::from_phrase(text.as_str(), Language::English) {
+        Ok(mnemonic) => mnemonic,
+        Err(_) => return Err(ClientKeyError::MnemonicFailure),
+    };
```

Comment:
Also, should we wrap the error in this case so we can send it back to the caller?

```suggestion
    let mnemonic = Mnemonic::from_phrase(text.as_str(), Language::English)
        .map_err(|_| ClientKeyError::MnemonicFailure)?;
```

Reply:
What do you mean by that?